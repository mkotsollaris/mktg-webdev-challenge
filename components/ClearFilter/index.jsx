import { useContext } from 'react'
import AppContext from '../../context/AppContext'
import style from '../style.module.css'
import componentsStyle from '../style.module.css'

const ClearFilter = () => {
  const { setFilteredDepartments } = useContext(AppContext)
  return (
    <button
      className={`${style['secondary-label']} ${style.clickable} ${componentsStyle['default-button']}`}
      onClick={() => setFilteredDepartments([])}
    >
      clear filters
    </button>
  )
}

export default ClearFilter
