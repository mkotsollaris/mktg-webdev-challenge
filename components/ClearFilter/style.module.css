.secondary-label {
  color: gray;
  font-size: small;
}
