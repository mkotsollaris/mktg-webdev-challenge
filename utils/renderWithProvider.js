import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
