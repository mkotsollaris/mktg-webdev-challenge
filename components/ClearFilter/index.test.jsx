import ClearFilter from '.'
import renderWithProvider from '../../utils/renderWithProvider'

it('renders', () => {
  const providerProps = {
    setFilteredDepartments: jest.fn(),
  }
  const { asFragment } = renderWithProvider(providerProps, ClearFilter)
  expect(asFragment()).toMatchSnapshot()
})
