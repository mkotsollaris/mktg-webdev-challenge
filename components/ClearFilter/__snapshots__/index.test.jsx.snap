// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="undefined undefined undefined"
  >
    clear filters
  </button>
</DocumentFragment>
`;
